### ISSUE TITLE
Mac Mini M2 Pro - `Found invalid cpu_subfamily: 0x00000004`

### ISSUE THREAD

**Issue by MrVeink** [Dr-Noob/cpufetch#189]
### Using a Mac Mini with an M2 Pro CPU with 10 CPU cores and 16 GPU cores. OS version is Ventura 13.4.

`% cpufetch --verbose`
```
[ERROR]: Found invalid cpu_subfamily: 0x00000004
[VERSION]: cpufetch v1.04 (macOS ARM build)
Please, create a new issue with this error message, your smartphone/computer model, the output of 'cpufetch --verbose' and 'cpufetch --debug' on https://github.com/Dr-Noob/cpufetch/issues
[WARNING]: SoC detection failed using cpu_subfamily

                                                            SoC:                 Unknown
                                                            Technology:          Unknown
                                                            CPU 1:
                                                              Microarchitecture: Blizzard
                                                              Max Frequency:     2.800 GHz
                                                              Cores:             4 cores
                                                              Features:          NEON,SHA1,SHA2,AES,CRC32
                                                            CPU 2:
                                                              Microarchitecture: Avalanche
                                                              Max Frequency:     3.500 GHz
                                                              Cores:             4 cores
                                                              Features:          NEON,SHA1,SHA2,AES,CRC32
                                                            Peak Performance:    100.80 GFLOP/s

```

`% cpufetch --debug`
```
[ERROR]: Found invalid cpu_subfamily: 0x00000004
[VERSION]: cpufetch v1.04 (macOS ARM build)
Please, create a new issue with this error message, your smartphone/computer model, the output of 'cpufetch --verbose' and 'cpufetch --debug' on https://github.com/Dr-Noob/cpufetch/issues
cpufetch v1.04 (macOS ARM build)
```

**Comment by MrVeink** [comment]
### Just noticed that compiling from source has the latest commit. Still getting an error, but a different one.

`./cpufetch --verbose`
```
[ERROR]: fill_soc: soc->soc_vendor == SOC_VENDOR_UNKOWN
[VERSION]: cpufetch v1.04-4-g3607 (macOS ARM build)
Please, create a new issue with this error message, your smartphone/computer model, the output of 'cpufetch --verbose' and 'cpufetch --debug' on https://github.com/Dr-Noob/cpufetch/issues
[WARNING]: SoC detection failed using cpu_subfamily

                                                            SoC:                 Unknown
                                                            Technology:          Unknown
                                                            CPU 1:
                                                              Microarchitecture: Blizzard
                                                              Max Frequency:     2.800 GHz
                                                              Cores:             6 cores
                                                              Features:          NEON,SHA1,SHA2,AES,CRC32
                                                            CPU 2:
                                                              Microarchitecture: Avalanche
                                                              Max Frequency:     3.500 GHz
                                                              Cores:             4 cores
                                                              Features:          NEON,SHA1,SHA2,AES,CRC32
                                                            Peak Performance:    716.80 GFLOP/s

```

`./cpufetch --debug`
```
[ERROR]: fill_soc: soc->soc_vendor == SOC_VENDOR_UNKOWN
[VERSION]: cpufetch v1.04-4-g3607 (macOS ARM build)
Please, create a new issue with this error message, your smartphone/computer model, the output of 'cpufetch --verbose' and 'cpufetch --debug' on https://github.com/Dr-Noob/cpufetch/issues
cpufetch v1.04-4-g3607 (macOS ARM build)
```

**Comment by Dr-Noob** [comment]
Thanks for reporting! Should be fixed in 6a5b4c3. Can you confirm in your machine?

**Comment by MrVeink** [comment]
`% cpufetch`
``` 

                   ."c.         
                 ,xNMM.         
                .lMM"           SoC:                 Apple M2 Pro
                MM*             Technology:          5nm
     .;loddo;:.   olloddol;.    CPU 1:
   cKMMMMMMMMMMNWMMMMMMMMMMM0:    Microarchitecture: Blizzard
 .KMMMMMMMMMMMMMMMMMMMMMMMW*      Max Frequency:     2.800 GHz
 XMMMMMMMMMMMMMMMMMMMMMMMX.       Cores:             6 cores
;MMMMMMMMMMMMMMMMMMMMMMMM:        Features:          NEON,SHA1,SHA2,AES,CRC32
:MMMMMMMMMMMMMMMMMMMMMMMM:      CPU 2:
.MMMMMMMMMMMMMMMMMMMMMMMMX.       Microarchitecture: Avalanche
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.     Max Frequency:     3.500 GHz
 'XMMMMMMMMMMMMMMMMMMMMMMMMMMk    Cores:             4 cores
  'XMMMMMMMMMMMMMMMMMMMMMMMMK.    Features:          NEON,SHA1,SHA2,AES,CRC32
    kMMMMMMMMMMMMMMMMMMMMMMd    Peak Performance:    716.80 GFLOP/s
     'KMMMMMMMWXXWMMMMMMMk.     
       "cooc"*    "*coo'"       


```

Seems to work great. Thanks!!
